src/AI/WallyAI.java

```diff
@@ -166,29 +224,456 @@ public void endTurn()
     log(String.format("[======== Wally ending turn %s for %s =========]", turnNum, myArmy));
   }
 
-  public static class SiegeAttacks extends UnitActionFinder
+  private void updatePlan(Object whodunit, int score, Unit unit, GamePath path, GameAction action)
+  {
+    updatePlan(whodunit, score, unit, path, action, false, 42);
+  }
+  private void updatePlan(Object whodunit, int score, UnitContext uc, GamePath path, GameAction action)
+  {
+    updatePlan(whodunit, score, uc, path, action, false, 42);
+  }
+  private void updatePlan(Object whodunit, int score, Unit unit, GamePath path, GameAction action, boolean isAttack, int percentDamage)
+  {
+    updatePlan(whodunit, score, new UnitContext(unit), path, action, isAttack, percentDamage);
+  }
+  private void updatePlan(Object whodunit, int score, UnitContext uc, GamePath path, GameAction action, boolean isAttack, int percentDamage)
+  {
+    if( null == action )
+      return;
+    final ActionPlan plan = new ActionPlan(whodunit, uc, action, score);
+    plan.path = path;
+    plan.isAttack = isAttack;
+    plan.percentDamage = percentDamage;
+    updatePlan(plan);
+  }
+  private void updatePlan(ActionPlan plan)
+  {
+    XYCoord dest = plan.action.getMoveLocation();
+    final UnitPrediction destPredictTile = mapPlan[dest.x][dest.y];
+    ActionPlan evicted = destPredictTile.toAchieve;
+    ActionPlan actorPrevPlan = plansByUnit.get(plan.actor.unit);
+    cancelPlan(evicted);
+    cancelPlan(actorPrevPlan);
+
+    // Assuming our path has been planned well re:terrain, running into enemies is the only obstacle
```

Comment:
This comment references enemies, but the code below does not. It just checks that the generated path isn't too short to be valid (also not sure how you are generating paths such that they might be too short to reach the destination. The comment is also repeated further down, so I suspect pasta.

Reply:
Seems both lines were added at the same time and it was inaccurate at that time (bd0caac), so agreed. Good catch.